README.md

```diff
@@ -1,707 +1,1799 @@
+
+| Name                          | Description                                                                                                                                                                                            |
+| :---------------------------- | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `InputSelectors`              | The type of the [`input selectors`] array.                                                                                                                                                             |
+| `Result`                      | The return type of the [`combiner`] as well as the [`output selector`].                                                                                                                                |
+| `OverrideMemoizeFunction`     | The type of the optional `memoize` function that could be passed into the options object to override the original `memoize` function that was initially passed into [`createSelectorCreator`].         |
+| `OverrideArgsMemoizeFunction` | The type of the optional `argsMemoize` function that could be passed into the options object to override the original `argsMemoize` function that was initially passed into [`createSelectorCreator`]. |
+
+</details>
+
+<div align="right">[ <a href="#table-of-contents">↑ Back to top ↑</a> ]</div>
+
+---
+
+### Memoization Functions
+
+<a id="defaultmemoize"></a>
+
+#### defaultMemoize(func, equalityCheckOrOptions = defaultEqualityCheck)
+
+<details><summary><b>Description</b></summary>
+
+The standard memoize function used by [`createSelector`].
+
+It has a default cache size of 1. This means it always recalculates when the value of an argument changes. However, this can be customized as needed with a specific max cache size (**`Since`** 4.1.0).
+
+It determines if an argument has changed by calling the `equalityCheck` function. As `defaultMemoize` is designed to be used with immutable data, the default `equalityCheck` function checks for changes using [`reference equality`][Reference Equality Check]:
+
+```ts
+const defaultEqualityCheck = (previousValue: any, currentValue: any) => {
+  return previousValue === currentValue
+}
+```
+
+</details>
+
+<details><summary><b>Parameters</b></summary>
+
+| Name                     | Description                                                 |
+| :----------------------- | :---------------------------------------------------------- |
+| `func`                   | The function to be memoized.                                |
+| `equalityCheckOrOptions` | Either an `equality check` function or an `options` object. |
+
+**`Since`** 4.1.0, `defaultMemoize` also accepts an options object as its first argument instead of an `equalityCheck` function. The `options` object may contain:
```

Comment:
```suggestion
Since 4.1.0, `defaultMemoize` also accepts an options object as its first argument instead of an `equalityCheck` function. The `options` object may contain:
```